### ISSUE TITLE
Making Change in The Backbone for Skeleton Based Action Recognition

### ISSUE THREAD

**Issue by taliegemen** [open-mmlab/mmaction2#1948]
Hello,
I am currently running slowonly_r50_u48_240e_ntu120_xsub_keypoint, the question I wanted to ask is can I make changes in the backbone (resnet50) of the training (such as using channel normalization instead of batch normalization)? If i can, is there any guide that I can use of (or any names for files that I should edit)?
Thanks!

**Comment by cir7** [comment]
Hi, @taliegemen , it depends on what changes you want to make. If the original backbone support the change in init function, modify the config of backbone module is the most convenient. Otherwise you may need to add a new backbone, follow this [guide](https://github.com/open-mmlab/mmaction2/blob/master/docs/tutorials/5_new_modules.md#add-new-backbones).   
And I am not sure what do you mean by `channel normalization`

**Comment by taliegemen** [comment]
I miswrite layer normalization with channel normalization for some reason, sorry for that. I will follow the guide you provided and update this post once I complete the guide.
Thanks a lot!

**Comment by cir7** [comment]
Hi, @taliegemen , actually, we support specify normalization layer type in `ResNet3d`. you can modify backbone config as follows:
`backbone=dict(
        type='ResNet3dSlowOnly',
        norm_cfg=dict(type='LN')
        ...)`
but i found some problem when try to use LayerNorm,
1. there is some hard-code in `ResNet3d:init_weights`, specify  normalization layer as batchnorm.
2. for img input, LayerNorm init argument is different to BatchNorm, which may cause a bug in our codebase.   

So i guess that you may need to build a custom backbone.

**Comment by taliegemen** [comment]
Thank you for your help, I should also ask is Group Normalization supported? Because I can modify Group Normalization parameters to make it just like layer normalization afaik.
Thanks a lot!

**Comment by cir7** [comment]
@taliegemen , yes, you are right, use `Group Normalization` will solve the second problem, and config is 
`backbone=dict(
        type='ResNet3dSlowOnly',
        norm_cfg=dict(type='GN', num_groups=1),
        ...)`
and for the first problem, you may need modify `.bn` to `.gn` in https://github.com/open-mmlab/mmaction2/blob/0ec7953ee3da629d6102f295cc3ba6591cc8032a/mmaction/models/backbones/resnet3d.py#L829-L831
and please note that directly modify the file will influence all ResNet3D model, will cause a error when use BN
